packages/components/src/color-picker/inputs.js

```diff
@@ -0,0 +1,287 @@
+				l: Number( ( data.l && data.l ) || this.props.hsl.l ),
+				source: 'hsl',
+			} );
+		}
+	}
+
+	renderFields() {
+		const { disableAlpha = false } = this.props;
+		if ( this.state.view === 'hex' ) {
+			return (
+				<fieldset>
+					<legend className="screen-reader-text">
+						{ __( 'Color value in HEX' ) }
+					</legend>
+					<div className="components-color-picker__inputs-fields">
+						<Input
+							label="hex"
+							value={ this.props.hex }
+							onChange={ this.handleChange }
+						/>
+					</div>
+				</fieldset>
+			);
+		} else if ( this.state.view === 'rgb' ) {
+			return (
+				<fieldset>
+					<legend className="screen-reader-text">
+						{ __( 'Color value in RGB' ) }
+					</legend>
+					<div className="components-color-picker__inputs-fields">
+						<Input
+							label="r"
+							value={ this.props.rgb.r }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="255"
+						/>
+						<Input
+							label="g"
+							value={ this.props.rgb.g }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="255"
+						/>
+						<Input
+							label="b"
+							value={ this.props.rgb.b }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="255"
+						/>
+						{ disableAlpha ? null : (
+							<Input
+								label="a"
+								value={ this.props.rgb.a }
+								onChange={ this.handleChange }
+								type="number"
+								min="0"
+								max="1"
+								step="0.05"
+							/>
+						) }
+					</div>
+				</fieldset>
+			);
+		} else if ( this.state.view === 'hsl' ) {
+			return (
+				<fieldset>
+					<legend className="screen-reader-text">
+						{ __( 'Color value in HSL' ) }
+					</legend>
+					<div className="components-color-picker__inputs-fields">
+						<Input
+							label="h"
+							value={ Math.round( this.props.hsl.h ) }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="359"
+						/>
+						<Input
+							label="s"
+							value={ Math.round( this.props.hsl.s * 100 ) }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="100"
+						/>
+						<Input
+							label="l"
+							value={ Math.round( this.props.hsl.l * 100 ) }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="100"
+						/>
+						{ disableAlpha ? null : (
+							<Input
+								label="a"
+								value={ this.props.hsl.a }
+								onChange={ this.handleChange }
+								type="number"
+								min="0"
+								max="1"
+								step="0.05"
+							/>
+						) }
+					</div>
+				</fieldset>
+			);
+		}
+	}
+
+	render() {
+		return (
+			<div className="components-color-picker__inputs-wrapper">
+				{ this.renderFields() }
+				<div className="components-color-picker__inputs-toggle">
+					<IconButton
+						icon="arrow-down-alt2"
+						label={ __( 'Toggle input type' ) }
```

Comment:
I'd consider to clarify this label. It certainly toggles the input type, but the verbiage seems a bit too technical to me. I'd suggest to use a text that clearly explains what it does, something along the lines of: "Toggle color format".

Reply:
Oops, yeah, @sarahmonster pointed that out to me in-person and I forgot. It's really not even a "toggle" between that implies a binary change; this actually goes through more than two colour selection options.

Reply:
So maybe something like "Change color picker format"?

Reply:
Yes, let's please swap out "toggle" for "change" here! (I think I mentioned this in overall comments.)

"Change color mode" might be more accurate here, although keeping in mind it's likely that a majority of users won't know what colour modes are.

Reply:
```suggestion
						label={ __( 'Change color mode' ) }
```

Reply:
Fixed at 4f82bae